### PR TITLE
[GCI-241] Synchronized methods in L247 and L269 of Context.java

### PR DESCRIPTION
**Pull request by johncgci**
## Issue I worked on

see https://issues.openmrs.org/browse/GCI-241
## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/context/Context.java [hunk 1]
```diff
@@ -266,7 +266,7 @@ static ServiceContext getServiceContext() {
 	 *
 	 * @param ctx
 	 */
-	public void setServiceContext(ServiceContext ctx) {
+	public void synchronized setServiceContext(ServiceContext ctx) {
```
Comment (ivange94) [inline api/src/main/java/org/openmrs/api/context/Context.java]: synchronized should be before void. Seems you didn't run `mvn clean package` after making this change else the compiler would have caught it. See travis error here https://travis-ci.org/openmrs/openmrs-core/builds/320193916#L2148

Comment (johncgci) [reply api/src/main/java/org/openmrs/api/context/Context.java]: I saw the error but didn't understand it. I thought it had to do with installation of plugins, and incorrectly ignored it. I have corrected my commit.

**Comment by dkayiwa** [comment]
Can you squash these commits into one as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by dkayiwa** [comment]
Did you test this by trying to run an openmrs instance with these changes?

**Pull request closed without merging**